src/deploy/functions/params.ts

```diff
@@ -248,6 +249,15 @@ export async function resolveParams(
   }
 
   for (const param of outstanding) {
+    if (nonInteractive) {
```

Comment:
Rather than have a developer iterate through `firebase deploy` calls until one finally succeeds, let's move this outside of the for loop and include all missing params in one error. Something like:

```
if (outstanding.length && nonInteractive) {
  const envNames = outstanding.map(p => p.name).join(", ");
  throw new FirebaseError(
    `In non-interactive mode but have no value for the following environment variables: ${envNames}\n` +
    "To continue, either run `firebase deploy` with an interactive terminal, or add values to a dotenv file. " +
    "For information regarding how to use dotenv files, see https://firebase.google.com/docs/functions/config-env"
  );
}

Reply:
Done, good call.